src/main/cluster.ts

```diff
@@ -263,9 +292,9 @@ export class Cluster implements ClusterModel {
 
   protected async getConnectionStatus(): Promise<ClusterStatus> {
     try {
-      const response = await this.k8sRequest("/version")
-      this.version = response.gitVersion
-      this.failureReason = null
+      const versionDetector = new VersionDetector(this)
+      await versionDetector.detect()
+      this.version = versionDetector.value
```

Comment:
why do we need to set `this.version`? should we just have `get version()` that reads value from metadata?

Reply:
version is now read from metadata